src/main/cljc/jarl/builtins/units.cljc

```diff
@@ -0,0 +1,194 @@
+(ns jarl.builtins.units
+  (:require [clojure.string :as str]
+            [jarl.builtins.utils :refer [numeric?]]
+            [jarl.exceptions :as errors]
+            #?(:cljs  [jarl.types :as types])
+            #?(:cljs [cljs.math :as math])))
+
+#?(:clj
+   (def unit-map
+     {""    (bit-shift-left 1 (* 10 0))
+      "ki"  (bit-shift-left 1 (* 10 1))
+      "kib" (bit-shift-left 1 (* 10 1))
+      "mi"  (bit-shift-left 1 (* 10 2))
+      "mib" (bit-shift-left 1 (* 10 2))
+      "gi"  (bit-shift-left 1 (* 10 3))
+      "gib" (bit-shift-left 1 (* 10 3))
+      "ti"  (bit-shift-left 1 (* 10 4))
+      "tib" (bit-shift-left 1 (* 10 4))
+      "pi"  (bit-shift-left 1 (* 10 5))
+      "pib" (bit-shift-left 1 (* 10 5))
+      "ei"  (bit-shift-left 1 (* 10 6))
+      "eib" (bit-shift-left 1 (* 10 6))
+
+      "k"   1000
+      "kb"  1000
+      "m"   (* 1000 1000)
+      "mb"  (* 1000 1000)
+      "g"   (* 1000 1000 1000)
+      "gb"  (* 1000 1000 1000)
+      "t"   (* 1000 1000 1000 1000)
+      "tb"  (* 1000 1000 1000 1000)
+      "p"   (* 1000 1000 1000 1000 1000)
+      "pb"  (* 1000 1000 1000 1000 1000)
+      "e"   (* 1000 1000 1000 1000 1000 1000)
+      "eb"  (* 1000 1000 1000 1000 1000 1000)
+
+      "siMilli" 0.001
+      "siK"     1000
+      "siM"     (* 1000 1000)
+      "siG"     (* 1000 1000 1000)
+      "siT"     (* 1000 1000 1000 1000)
+      "siP"     (* 1000 1000 1000 1000 1000)
+      "siE"     (* 1000 1000 1000 1000 1000 1000)}))
+
+#?(:cljs
+  (def unit-map
+    {""     (bit-shift-left (js/BigInt 1) (* (js/BigInt 10) (js/BigInt 0)))
```

Comment:
Could we instead have our own `bit-shift-left` and `*` functions that differentiate between `clj` and `cljs`, to avoid having these two big `unit-map`s?

Reply:
Or maybe just an `convert-int` function, or something used for wrapping each value.

Reply:
Might come back to do this, or even decouple the bigint handling to a separate namespace... but honestly so tired of it right now, lol 😄